six/modules/c++/cphd/include/cphd/PVP.h

```diff
@@ -396,22 +415,36 @@ struct Pvp
     size_t getAdditionalParamsSize() const;
 
     /*
-     *  \func setData
+     *  \func specifyOptionalParameters
+     *  \brief Marks optional parameters flags
      *
-     *  \brief Validate and set the metadata of the specified parameter
+     *  \param hasAmpSF flag indicates if optional param AmpSF exists
+     *  \param hasFxN1 flag indicates if optional param FxN1 exists
+     *  \param hasFxN2 flag indicates if optional param FxN2 exists
+     *  \param hasToaE1 flag indicates if optional param ToaE1 exists
+     *  \param hasToaE2 flag indicates if optional param ToaE2 exists
+     *  \param hasTDIonoSRP flag indicates if optional param TDIonoSRP exists
+     *  \param hasSignal flag indicates if optional param signal exists
+     */
+    void specifyOptionalParameters(bool hasAmpSF = false, bool hasFxN1 = false, bool hasFxN2 = false,

```

Comment:
Now that I actually see it, I think having a function that takes 7 boolean arguments is going to be hard to read and easy to misuse. Can you find a more explicit interface, a separate function for each parameter or passing flags, or something?

(And yeah, I realize this was my bad idea...)